src/copycutter/text_question.py

```diff
@@ -349,41 +372,40 @@ def call_cookie_template(self) -> None:
                 if select.value[-1] is not None:
                     # Again, while choosing defaults checks if the select-field is represented
                     # as a `dict`
-                    context[str(select.value[2])] = select.value[-1][
-                        select._input._options[1][0]
-                    ]
+                    value = select.value[-1][list(select._input._options)[1][0]]
+                    context[str(select.value[2])] = value
                 else:
-                    context[str(select.value[2])] = select._input._options[1][0]
+                    context[str(select.value[2])] = list(select._input._options)[1][0]
```

Comment:
This might be an issue with Textual's typing - it claims it returns an Iterable. You should always return concrete types, Protocols are for input variables, not returns.

If it really does return Iterables instead of Sequences/lists/tuples, then the list is actually required here.